docs/design/constant-propagation-and-branch-removal.md

```diff
@@ -64,7 +64,7 @@ The implementation of this optimization is relatively complex since it's solving
   * Sentinel value "Processed and is not constant" which means the method has been processed and its return value was not detected as constant. This is a final state.
   * Instruction which represents the constant return value of the method if it was detected as returning constant value. This is a final state.
 
-* Processing stack which stores ordered list of processing node, each node representing a method and additional data about it. The stack is processed by always taking the top of the stack and attempting to process that node. Nodes are always added to the top of the stack and are always removed from the top of the stack. In some cases nodes are "moved", that is a node which is not on the top of the stack is moved to the top of the stack. For this reason the stack is implemented as a linked list (so that it's easy to point to nodes in it as well as moves nodes around).
+* Processing stack which stores ordered list of processing node, each node representing a method and additional data about it. The stack is processed by always taking the top of the stack and attempting to process that node. Nodes are always added to the top of the stack and are always removed from the top of the stack. In some cases nodes are "moved", that is a node which is not on the top of the stack is moved to the top of the stack. For this reason the stack is implemented as a trimmed list (so that it's easy to point to nodes in it as well as moves nodes around).
```

Comment:
"linked list" :)
```suggestion
* Processing stack which stores ordered list of processing node, each node representing a method and additional data about it. The stack is processed by always taking the top of the stack and attempting to process that node. Nodes are always added to the top of the stack and are always removed from the top of the stack. In some cases nodes are "moved", that is a node which is not on the top of the stack is moved to the top of the stack. For this reason the stack is implemented as a linked list (so that it's easy to point to nodes in it as well as moves nodes around).
```